internal/provider/providertest/utils.go

```diff
@@ -157,6 +157,30 @@ func StateHasExpectedValues(resourceName string, expected map[string]any) resour
 	}
 }
 
+func StateDoesNotHaveFields(resourceName string, fields []string) resource.TestCheckFunc {
```

Comment:
Used this to verify that fields not set by the user (null) are omitted from the TF state